### PR TITLE
Kata2 tests

### PR DESCRIPTION
**Pull request by vitvizel**
kata2 tests no final

### CODE REVIEW

File: python_katas/kata_2/test.py [hunk 1]
```diff
@@ -71,114 +122,223 @@ def test_sample(self):
         pass
```
Comment (DannyAtal) [inline python_katas/kata_2/test.py:122]: also about the TestJsonConfigsMerge, this one no one worked on it, we can if you like.
ask the group if someone want to join and we will build it.

File: python_katas/kata_2/test.py [hunk 2]
```diff
@@ -12,43 +12,94 @@
...
     def test_sample(self):
-        # your code here
-        pass
+            number = 6
+            self.assertEqual(questions.fibonacci_fixme(number), 8)
+
+            number = 5
+            self.assertEqual(questions.fibonacci_fixme(number), 5)
 
+            number = 0
```
Comment (ronhadad22) [inline python_katas/kata_2/test.py:45]: no need to check 0

File: python_katas/kata_2/test.py [hunk 3]
```diff
@@ -12,43 +12,94 @@
...
+       return file_path
+
+   def test_out_of_folder_directory(self):
+       file_path = "names2.txt"
+       self.assertEqual(questions.most_frequent_name(file_path), "Most repeated name: danny")
+       return file_path
+
+   def test_in_folder_directory(self):
+       file_path = "names.txt"
+       self.assertEqual(questions.most_frequent_name(file_path), "Most repeated name: Tawsha")
```
Comment (ronhadad22) [inline python_katas/kata_2/test.py:65]: no need to return the prefix "most repeated name"
only the name need to return

File: python_katas/kata_2/test.py [hunk 4]
```diff
@@ -12,43 +12,94 @@
...
-        # your code here
-        pass
 
+
+
+   def most_frequent_name(file_path):
+       return file_path
+
+   def test_out_of_folder_directory(self):
+       file_path = "names2.txt"
```
Comment (ronhadad22) [inline python_katas/kata_2/test.py:59]: there is no file "names2.txt"

File: python_katas/kata_2/test.py [hunk 5]
```diff
@@ -12,43 +12,94 @@
...
-        # your code here
-        pass
+    def test_infolder_directory(self):
+        dir_path = "files_to_backup"
+        self.assertEqual(questions.files_backup(dir_path), "files_to_backup")
+        return dir_path
+
+    def test_out_of_folder_directory(self):
+        dir_path = "/Users/gil/katas/DevOpsSep22/python_katas"
+        self.assertEqual(questions.files_backup(dir_path), "/Users/gil/katas/DevOpsSep22/python_katas")
```
Comment (ronhadad22) [inline python_katas/kata_2/test.py:91]: the func return  'backup_<dir_name>_<yyyy-mm-dd>.tar.gz' only

File: python_katas/kata_2/test.py [hunk 6]
```diff
@@ -71,114 +122,223 @@ def test_sample(self):
...
+        self.assertEqual(questions.monotonic_array([ ])), True)
+
+    def test_zero_monotonic_array(self):
+        self.assertEqual(questions.monotonic_array([0])), True)
+
+    def test_negative_num_monotonic_array(self):
+        self.assertEqual(questions.monotonic_array([[-1, -3, -5]])), True)
+
+    def self.test_negative_and_positive_monotonic_array(self):
+            self.assertEqual(questions.monotonic_array([[[-10, -5, 11, 1000]]])), True)
```
Comment (ronhadad22) [inline python_katas/kata_2/test.py:142]: good job

File: python_katas/kata_2/test.py [hunk 7]
```diff
@@ -71,114 +122,223 @@ def test_sample(self):
...
-    def test_sample(self):
-        # your code here
-        pass
+    def test_CapitalwithRegular_strings(self):
+        str1, str2 = ('AbbZD', 'abZd')
+    self.assertEqual(questions.longest_common_substring(str1, str2), bZ)
 
+    def test_empty_strings(self):
+        str1, str2 = ('a2a', 'a2a')
+    self.assertEqual(questions.longest_common_substring(str1, str2), a)
```
Comment (ronhadad22) [inline python_katas/kata_2/test.py:202]: why no a2a
and why no quata ""

File: python_katas/kata_2/test.py [hunk 8]
```diff
@@ -71,114 +122,223 @@ def test_sample(self):
...
 
+            str1 = 'Wonder'
+            str2 = 'Wonderful'
+            self.assertEqual(questions.longest_common_prefix(str1, str2), "Wonder")
 
+            str1 = "What's going on"
+            str2 = 'Hi everybody'
+            self.assertEqual(questions.longest_common_prefix(str1, str2), "")
+
+#Shani
```
Comment (ronhadad22) [inline python_katas/kata_2/test.py:227]: great job

File: python_katas/kata_2/test.py [hunk 9]
```diff
@@ -71,114 +122,223 @@ def test_sample(self):
...
+        self.assertEqual(questions.pascal_triangle(lines), 100)
 
+    def test_zero_as_input(self):
+        lines = 0
+        self.assertIsNotNone(questions.pascal_triangle(lines))
+
+    def test_no_input(self):
+        lines = ''
+        self.assertRaises(ValueError, questions.pascal_triangle, lines)
+
```
Comment (ronhadad22) [inline python_katas/kata_2/test.py:291]: the return to this func is None

File: python_katas/kata_2/test.py [hunk 10]
```diff
@@ -71,114 +122,223 @@ def test_sample(self):
...
+        self.assertEqual(questions.strong_pass(password), False)
+    def NoUpperCase(self):
+        password = 'tja!@jq7@#'
+        self.assertEqual(questions.strong_pass(password), False)
+    def NoSpecialChar(self):
+        password = 'ASDrh12647'
+        self.assertEqual(questions.strong_pass(password), False)
+    def StrongPassFTW(self):
+        password = 'A@Drh12$47'
+        self.assertEqual(questions.strong_pass(password), True)
```
Comment (ronhadad22) [inline python_katas/kata_2/test.py:339]: great

**Review by ronhadad22 — CHANGES_REQUESTED** [review]

File: python_katas/kata_2/test.py (hunk 3 above)
Comment (DannyAtal) [reply python_katas/kata_2/test.py:65]: done

File: python_katas/kata_2/test.py (hunk 4 above)
Comment (DannyAtal) [reply python_katas/kata_2/test.py]: done

File: python_katas/kata_2/test.py (hunk 9 above)
Comment (DannyAtal) [reply python_katas/kata_2/test.py:299]: done

**Review by DannyAtal — COMMENTED** [review]
@ronhadad22 can you please check the PR again?
I just pushed the last test.

**Review by ronhadad22 — APPROVED** [review]

**Pull request merged**